test-utils/state-viewer/src/apply_chain_range.rs

```diff
@@ -262,18 +297,45 @@ pub fn apply_chain_range(
                 if !smart_equals(&existing_chunk_extra, &chunk_extra) {
                     assert!(false, "Got a different ChunkExtra:\nblock_height: {}, block_hash: {}\nchunk_extra: {:#?}\nexisting_chunk_extra: {:#?}\nnew outcomes: {:#?}\n\nold outcomes: {:#?}\n", height, block_hash, chunk_extra, existing_chunk_extra, apply_result.outcomes, old_outcomes(store.clone(), &apply_result.outcomes));
                 }
-            },
+            }
             None => {
                 assert!(prev_chunk_extra.is_some());
                 assert!(apply_result.outcomes.is_empty());
                 if verbose_output {
                     println!("block_height: {}, block_hash: {}\nchunk_extra: {:#?}\nprev_chunk_extra: {:#?}\noutcomes: {:#?}", height, block_hash, chunk_extra, prev_chunk_extra, apply_result.outcomes);
                 }
-            },
+            }
         };
-        maybe_add_to_csv(&csv_file_mutex, &format!("{},{},{},{},{},{},{},{},{},{}", height, block_hash, block_author, num_tx, num_receipt, block.header().raw_timestamp(), apply_result.total_gas_burnt, chunk_present, apply_result.processed_delayed_receipts.len(), delayed_indices.map_or(0,|d|d.next_available_index-d.first_index)));
-        inc_and_report_progress(&processed_blocks_cnt, &processed_blocks_timestamp, all_blocks, &skipped_blocks);
-    });
+        maybe_add_to_csv(
+            &csv_file_mutex,
+            &format!(
+                "{},{},{},{},{},{},{},{},{},{}",
+                height,
+                block_hash,
+                block_author,
+                num_tx,
+                num_receipt,
+                block.header().raw_timestamp(),
+                apply_result.total_gas_burnt,
+                chunk_present,
+                apply_result.processed_delayed_receipts.len(),
+                delayed_indices.map_or(0, |d| d.next_available_index - d.first_index)
+            ),
+        );
+        inc_and_report_progress(
+            &processed_blocks_cnt,
+            &processed_blocks_timestamp,
+            all_blocks,
+            &skipped_blocks,
+        );
+    };
+    let range = (start_height..=end_height);
+
+    if cfg!(feature = "single_threaded") {
```

Comment:
It seems much simpler to pass a flag at runtime instead of recompiling with a different feature.

Reply:
I was thinking that we may want to run more functions with such option, and passing it everywhere would become a routine.

But it's not the case now, so you are right.